### PR TITLE
mypy: fix config

### PR DESCRIPTION
**Pull request by oliver-sanders**
Copy cylc config into rose.

### CODE REVIEW

**Review by MetRonnie — CHANGES_REQUESTED** [review]
It needs `types-aiofiles` in the `setup.py` dependencies or `mypy --install-types` twice in the GH Actions workflow. We went for the former in cylc-flow

**Review by MetRonnie — APPROVED** [review]
Leaving it to you to merge in case you want to add `allow_redefinition = True` as I've proposed in cylc-flow: https://github.com/cylc/cylc-flow/pull/4304

File: mypy.ini [hunk 1]
```diff
@@ -7,5 +7,8 @@ exclude = metomi/rose/etc/
 # Enable PEP 420 style namespace packages, which we use.
 # Needed for associating "import foo.bar" with foo/bar.py
 namespace_packages = True
+# Need this because we don't have __init__.py in top-level dir:
+explicit_package_bases = True
```
Comment (oliver-sanders) [inline mypy.ini:11]:
```suggestion
explicit_package_bases = True
allow_redefinition = True
```

**Comment by oliver-sanders** [comment]
Added

**Review by MetRonnie — APPROVED** [review]

**Review by wxtim — APPROVED** [review]

**Pull request merged**